### DISCUSSION TITLE
WRF HYDRO input redundancy

### DISCUSSION THREAD

**Discussion by jbensabat** [NCAR/wrf_hydro_nwm_public#732] [Q&A]
Hello 
I am running WRF HYDRO more or less as a novice and I have the following question:
The input files (nc files) contain data on parameters such as manning friction coefficient, soil properties etc.
but WRFHYDRO requires also to read ASCII files (*.TBL)  with overlapping information
what is the information that prevails ? the one in the nc files or the one in the TBL files ?

or in other words is there any documentation that explains how these files are used by WRFHYDRO

there are two motivations for my question: 1) a proper understanding of the input used by WRFHYDRO 2) I am working on coupling the parameter estimation pestpp (USGS) for sensitivity analysis and calibration. pestpp needs ASCII files so the TBL files are a good way to feed data to pestpp, but I want to make sure that I understand how the data in TBL files is used by WRFHYDRO

best
jac

**Comment by aubreyd** [comment]
We added this capability to specify a number of parameters as spatially variable (in hydro2dtbl.nc and soil_properties.nc) specifically for model calibration activities. If you specify an existing hydro2dtbl.nc file, those values will be used. If you provide the name of a file that does not exist, the code should read in HYDRO.TBL and create hydro2dtbl.nc for you. The Noah-MP parameters in soil_properties.nc will only be used if you compile the WRF-Hydro code with SPATIAL_SOIL = 1, otherwise SOILPARM.TBL and MPTABLE.TBL will be used. MPTABLE.TBL is always required, since it has many additional parameters that are not included in soil_properties.nc - we only included parameters in soil_properties.nc that we were commonly calibrating.

**Comment by jbensabat** [comment]
Thanks a lot for the clarification. 
in this subject could you please indicate which calibration software you are using ? as I said in my previous message we are preparing for pestpp.
best 
jac

**Comment by arezoorn** [comment]
Jac, 

We heavily use PyWrfHydroCalib toolkit which is using DDS as the optimization technique. We also use PESTPP for calibration, but we are at a research phase mostly with it, and it has not been used in our large scale calibration activities so far (for example NWM). As you mentioned PEST++ does not work with the netcdf files directly, so if you want to use the existing capabilities of the PEST++ you could compile the model with  SPATIAL_SOIL = 0, and then the parameters as Aubrey mentioned will be read from TBL files. However, this does not allow you to have spatial parameters in space which is a requirement for us when we do large scale modeling. Therefore, we are taking a different approach when coupling with PEST++ and adding post-processing script to the PEST++. It reads the parameters from a text file (template file) and writes it to netcdf format for WRF-Hydro use. You could do either of these two options. 

Thank you!
Arezoo

**Comment by jbensabat** [comment]
Hi and thanks for the info.
Actually i am preparing an application that can create template and ascii data files from a netcdf file and a netcdf  file from a data  file.
this would apply to 

route_link.nc
soil_properties.nc
lakeparm.nc
gwbuckparm.nc
hydro2tbl.nc 

I shall test python toolkit too. There is a another powerful package, specially developed for large scale applications: GOOGLE CERES
best 
jac
